### PR TITLE
Turn on cacheability for PitestTask

### PR DESCRIPTION
**Pull request by davidburstrom**
Fixes https://github.com/szpak/gradle-pitest-plugin/issues/67

### CODE REVIEW

File: src/funcTest/groovy/info/solidsoft/gradle/pitest/functional/PitestPluginGeneralFunctionalSpec.groovy [hunk 1]
```diff
@@ -112,6 +112,24 @@ class PitestPluginGeneralFunctionalSpec extends AbstractPitestFunctionalSpec {
...
+        writeHelloPitTest()
+        when:
+        ExecutionResult result = runTasksSuccessfully('pitest', '--build-cache')
+        ExecutionResult result2 = runTasksSuccessfully('clean', 'pitest', '--build-cache')
+        then:
+        result.wasExecuted(':pitest')
+        result.getStandardOutput().contains("Stored cache entry for task ':pitest'")
+        and:    //it works with @OutputFile by default, but just in case
+        result2.wasExecuted(':pitest')
+        result2.getStandardOutput().contains("Task :pitest FROM-CACHE")
```
Comment (szpak) [inline src/funcTest/groovy/info/solidsoft/gradle/pitest/functional/PitestPluginGeneralFunctionalSpec.groovy:130]: Nice. Just please reformat that test to be consistent with the others.

In Idea:
https://solidsoft.wordpress.com/2014/05/20/more-compact-automatic-spock-specification-formatting-in-idea-13/

**Review by szpak — CHANGES_REQUESTED** [review]

File: src/funcTest/groovy/info/solidsoft/gradle/pitest/functional/PitestPluginGeneralFunctionalSpec.groovy (hunk 1 above)
Comment (davidburstrom) [reply src/funcTest/groovy/info/solidsoft/gradle/pitest/functional/PitestPluginGeneralFunctionalSpec.groovy]: Thanks for the IDEA advice, I've updated it. I'm just very used to having automatic code style formatting and/or verification when I run `build` :)

**Review by szpak — APPROVED** [review]

**Comment by szpak** [comment]
With your PRs, I have a better motivation to resolve the issue with CI and release a new version :-).

**Comment by davidburstrom** [comment]
I'm happy to hear that. I think Pitest is one of the most underrated test tools in the JVM industry.

**Pull request merged**